ui/src/components/ThemeMode.tsx

```diff
@@ -0,0 +1,60 @@
+// This file is part of Moonfire NVR, a security camera network video recorder.
+// Copyright (C) 2021 The Moonfire NVR Authors; see AUTHORS and LICENSE.txt.
+// SPDX-License-Identifier: GPL-v3.0-or-later WITH GPL-3.0-linking-exception
+
+import { useColorScheme } from "@mui/material";
+import React, { createContext, useCallback } from "react";
+
+export enum CurrentMode {
+  Auto = 0,
+  Light = 1,
+  Dark = 2
+}
+
+interface ThemeProps {
+  changeTheme: () => void,
+  currentTheme: 'dark' | 'light',
+  choosenTheme: CurrentMode
+}
+
+export const ThemeContext = createContext<ThemeProps>({
+  currentTheme: window.matchMedia("(prefers-color-scheme: dark)").matches ? 'dark' : 'light',
+  changeTheme: () => { },
+  choosenTheme: CurrentMode.Auto
+});
+
+const ThemeMode = ({ children }: { children: JSX.Element }): JSX.Element => {
+  const { mode, setMode } = useColorScheme();
+
+  const useMediaQuery = useCallback((query: string) => {
+    const [matches, setMatches] = React.useState(
+      () => window.matchMedia(query).matches
+    );
+    React.useEffect(() => {
+      const m = window.matchMedia(query);
+      const l = () => setMatches(m.matches);
+      m.addEventListener("change", l);
+      return () => m.removeEventListener("change", l);
+    }, [query]);
+    return matches;
+  }, []);
+
+  const detectedSystemColorScheme = useMediaQuery("(prefers-color-scheme: dark)") ? "dark" : "light";
+
+  const changeTheme = React.useCallback(() => {
+    setMode(mode === 'dark' ? 'light' : mode === 'light' ? 'system' : 'dark')
+  }, [mode]);
```

Comment:
> `React Hook React.useCallback has a missing dependency: 'setMode'. Either include it or remove the dependency array`

Huh, whatever magic makes hook dependencies not need `setFoo` declared separately after `const [foo, setFoo] = useState(...)` doesn't seem to work with mui's `useColorScheme`. I guess just add `setMode` here?